### ISSUE TITLE
Non linear bar chart value axis

### ISSUE THREAD

**Issue by bragma** [apache/incubator-echarts#6817]
Hi,
I need to draw a single horizontal bar chart, but the values axis is not linear: it is comprised of 3 ranges:

[30,80)
[80,160)
[160, 580]

The bar must be broken down to 3 colors, depending on the part of the range it covers. This is an example made with google charts:

![image](https://user-images.githubusercontent.com/1101058/31438838-1f7670c8-ae8a-11e7-9e38-0774a5b40b72.png)

This is obtained with a fake setup, i.e. using a stacked bar chart with 3 different values for the 3 parts. This allows me to set colors independently, but the implementation is ugly.

I'd like to know if it is possible to obtain this with echarts in a cleaner way.

Thanks!

**Comment by Ovilia** [comment]
Hi Bragma, I'm afraid it can only be done in the same stacked way in ECharts.

**Comment by bragma** [comment]
@Ovilia 
thanks a lot, I supposed that. Another question: is it possible to "offset" or configure the ticks/labes aside from min, max and interval of the axes? If you look at the pic, you'll see that the first label is at value 40 and the next one at 60, so the step is 20. But the leftmost value (not shown) i.e. min is 30, not 20. I've not found a way to start ticks/labels in this way. As an examle, min:0, Interval: 20, but have the marks at 10, 30, 50, etc. instead of 20, 40, 60, etc.
The only solution I've found is to double the interval and using a formatter on the label to print a label every 2, but this leaves me with doubled ticks.
Thanks again!

**Comment by Ovilia** [comment]
It seems odd to me that before the number 100, the interval is 20, but after 100, the interval is 60. I wonder in which case you need this feature?

Here's a demo I made for you:
http://gallery.echartsjs.com/editor.html?c=xSJNEgnT2-&v=1

```
    xAxis:{
        interval: 10,
        min: 30,
        axisLabel: {
            formatter: function (v) {
                if (v > 30 && v % 20 === 0) {
                    return v;
                } 
            }
        },
        axisTick: {
            show: false
        },
        splitLine: {
            lineStyle: {
                color: ['transparent', '#aaa']
            }
        }
    },
```

**Comment by bragma** [comment]
The use case is to mimic a pre-printed document where several parts of a bar chart are "companded" to visually zoom into more interesting areas. Since the document will be printed it cannot be interacted.

This requires a trick before drawing the chart: it must be broken down into several stacked bars, all drawn on the same linear axis. The length of each bar must be scaled appropriately for the specific range of the main (linear) axis. It's ugly, but it works.

Your example (which I thank you for) is interesting because:

1. You supersampled the interval and used the formatter to place/hide labels, which is exactly the same trick I used. :)
1. You disabled ticks, since there is no way to "offset" them, as I suspected. I can live with it.

It would be nice to have an option to draw ticks only where a label is not null.

Thanks again!
